bindings/python/PyAnimateableProperty.cpp

```diff
@@ -0,0 +1,353 @@
+/*******************************************************************************
+ * Copyright 2009-2025 Jörg Müller
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+#include "PyAnimateableProperty.h"
+
+#include "Exception.h"
+
+#include "sequence/AnimateableProperty.h"
+
+#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
+#include <numpy/ndarrayobject.h>
+
+using namespace aud;
+
+extern PyObject* AUDError;
+
+static PyObject* AnimateableProperty_new(PyTypeObject* type, PyObject* args, PyObject* kwds)
+{
+	AnimateablePropertyP* self = (AnimateablePropertyP*) type->tp_alloc(type, 0);
+
+	int count;
+	float value;
+
+	if(self != nullptr)
+	{
+		if(!PyArg_ParseTuple(args, "i|f:animateableProperty", &count, &value))
+			return nullptr;
+
+		try
+		{
+			if(PyTuple_Size(args) == 1)
+			{
+				self->animateableProperty = new std::shared_ptr<aud::AnimateableProperty>(new aud::AnimateableProperty(count));
+			}
+			else
+			{
+				self->animateableProperty = new std::shared_ptr<aud::AnimateableProperty>(new aud::AnimateableProperty(count, value));
+			}
+		}
+		catch(aud::Exception& e)
+		{
+			Py_DECREF(self);
+			PyErr_SetString(AUDError, e.what());
+			return nullptr;
+		}
+	}
+
+	return (PyObject*) self;
+}
+
+static void AnimateableProperty_dealloc(AnimateablePropertyP* self)
+{
+	if(self->animateableProperty)
+		delete reinterpret_cast<std::shared_ptr<aud::AnimateableProperty>*>(self->animateableProperty);
+	Py_TYPE(self)->tp_free((PyObject*) self);
+}
+
+static PyObject* AnimateableProperty_read(AnimateablePropertyP* self, PyObject* args)
+{
+	float position;
+
+	if(!PyArg_ParseTuple(args, "f", &position))
+		return nullptr;
+
+	int count = (*reinterpret_cast<std::shared_ptr<aud::AnimateableProperty>*>(self->animateableProperty))->getCount();
+	npy_intp dims[1] = {count};
+	PyObject* np_array = PyArray_SimpleNew(1, dims, NPY_FLOAT32);
+	if(!np_array)
+		return nullptr;
+
+	float* out = (float*) PyArray_DATA((PyArrayObject*) np_array);
```

Comment:
Please use `reinterpret_cast`, also for the other C style casts further down below.